### DISCUSSION TITLE
Developing custom block based theme vs sticking with understrap

### DISCUSSION THREAD

**Discussion by anderskitson** [holger1411/understrap#1888] [Q&A] [answered]
Deciding if I should go full site editing theme, or use understrap, What are the benefit of sticking with understrap? The team I am working with has familiarity with understrap me less so, we are developer heavy and it seems full site editing is focused highly on building the site in the editor visually. Are there things besides this that would make understrap a better choice than developing a true block based theme?

**Comment by bacoords** [comment, marked as answer]
Here's my personal take on this. Know that this very much depends on the expectations, time, and budget of every individual project.

If you are
- interested in designing and building your site in the browser (similar to using something like Elementor)
- OK with relying on third-party builder tools like Bricky, Coblocks, or Kadence to fill in the gaps
- not worried about matching a 'pixel perfect' design mockup exactly
- not adding anything advanced like a custom API integration or animations
- comfortable using React to build custom blocks should the need arise
- OK with knowing that some functionality may break or change over time
- OK with not having any version control (other than full site backups) as you build out your design
- building a simple landing page or single-column blog

...then you can probably get away with using FSE. I still wouldn't consider it 'production-ready' for most advanced sites. Basically  you'll still need to do a lot of custom coding still to get things working, but you'll be also stuck with the HTML generated by Gutenberg unless you plan on building custom React blocks. But for a simple blog or brochure site, you could potentially make it work.

If you are
- comfortable writing your own code for the front-end of your website in a code editor
- very concerned about matching pixel-perfect designs
- working with end users who might get confused/overwhelmed by an overly visual interface like the site editor
- adding anything custom or advanced like a mega menu, API integration, e-commerce, membership plugins, interactive or animated elements, etc
- concerned about having total control over HTML/CSS/JS for performance or accessibility goals
- planning to have any templates that you want to be able to make global updates on (say like a very specific custom post type)

Then I'd suggest sticking with Understrap or a similar 'classic/hybrid' theme. I just don't see FSE themes being versatile enough to compete with custom coding your website. 

That said, I still like using the block editor for much of the actual site content, especially with a tool like ACF Blocks that allows you to quickly add custom code inside the block editor. Understrap has strong support for the block editor (and the upcoming release will extend this even further)  All of this depends on your particular use case. 

Hope this helps and I'm hopeful that others will chime in with their experiences.

**Reply by anderskitson** [reply]
Thanks for this detailed answer it helps a lot. I did play with FSE a bit and it did not make that much sense for our build, since I think we will need to do a lot of custom coding still. This was super helpful and a great reference for the main differences between the two as of now. Thanks again.
